src/iso8583/mli.rs

```diff
@@ -77,6 +90,15 @@ impl MLI for MLI4E {
         let _ = mli.write_u32::<byteorder::BigEndian>(n.clone() as u32);
         Ok(mli)
     }
+
+    fn is_available(&self, stream: &TcpStream) -> Result<bool, IsoError> {
+        let mut buf = vec![0; 4];
+
+        match stream.peek(&mut buf).unwrap() {
```

Comment:
The unwrap could panic, so something like this should do?

```rust
match stream.peek(&mut buf) {
                    Ok(n) => {
                        if n==4{
                            Ok(true)
                        }else{
                            Err(IsoError { msg: format!("client disconnected")})
                        }
                    },
                    Err(e) => Err(IsoError { msg: format!("stream err. cause: {}",e.to_string())})
                }
```

Reply:
Excellent solution!  Thanks for showing me that, I will be using this pattern often in the future.